jcl/src/openj9.jvm/share/classes/com/ibm/jvm/CRIUSupport.java

```diff
@@ -43,4 +77,49 @@ public static boolean isCRIUSupportEnabled() {
 		return criuSupportEnabled;
 	}
 	
+	/**
+	 * Checkpoint the JVM
+	 * 
+	 * @param cpDataDir the directory in which the checkpoint data will be stored
+	 * 
+	 * @return return CRIUOpRetCode
+	 */
+	public static CRIUOpRetCode checkPointJVM(String cpDataDir) {
```

Comment:
This seems like a security problem, allowing unrestricted access to write anywhere. There should be a permission check added.